### PR TITLE
Optimization: Lyrics Parser Restructure

### PR DESCRIPTION
**Pull request by ZhongYic00**
按照注释部分要求用正则表达式重写lrc文件解析。
抽象出parseLyric方法以方便网络下载等来源的歌词解析使用。
顺便更正拼写错误。
望采纳

### CODE REVIEW

File: src/music-player/core/util/musiclyric.cpp [hunk 1]
```diff
@@ -90,13 +91,36 @@ static QString getFileCodex(QString dir)
...
+        qint64 time = t.msecsSinceStartOfDay();
+        if(t.isValid())
+            tmp.push_back({time,lyricStr});
+    }
+    std::sort(tmp.begin(),tmp.end());
+    this->position.clear();
+    this->line.clear();
+    for(auto item:tmp) {
+        this->position.push_back(item.first);
+        this->line.push_back(item.second);
```
Comment (kt286) [inline src/music-player/core/util/musiclyric.cpp:113]: `this->line.push_back(item.second.trimmed());`

see https://github.com/linuxdeepin/deepin-music/pull/112

Comment (ZhongYic00) [reply src/music-player/core/util/musiclyric.cpp:113]: or we can fix the RegExp to `\\[([^\\]]*)\\]\\s*(\\S.*\\S)\\s*$`
which works as
image

Comment (kt286) [reply src/music-player/core/util/musiclyric.cpp:113]: cool

**Comment by lzwind** [comment]
There is a conflict in the current assembly line. Please resubmit it. I'm very sorry for the trouble caused to you.

**Comment by ZhongYic00** [comment]
@lzwind early commit messages conflict with your PR standard, do I need to restart another pull request?

**Comment by kt286** [comment]
It is recommended to close and submit a new PR

**Pull request closed without merging**